src/main/scala/io/iohk/ethereum/blockchain/sync/Blacklist.scala

```diff
@@ -35,60 +34,143 @@ object Blacklist {
         val code: Int = 2
         val name: String = "BlockHeaderValidationFailed"
       }
-      case object ErrorInBlockHeadersType extends BlacklistReasonType {
+      case object ErrorInBlockHeadersType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 3
         val name: String = "ErrorInBlockHeaders"
       }
-      case object EmptyBlockBodiesType extends BlacklistReasonType {
+      case object EmptyBlockBodiesType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 4
         val name: String = "EmptyBlockBodies"
       }
-      case object BlockBodiesNotMatchingHeadersType extends BlacklistReasonType {
+      case object BlockBodiesNotMatchingHeadersType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 5
         val name: String = "BlockBodiesNotMatchingHeaders"
       }
-      case object EmptyReceiptsType extends BlacklistReasonType {
+      case object EmptyReceiptsType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 6
         val name: String = "EmptyReceipts"
       }
-      case object InvalidReceiptsType extends BlacklistReasonType {
+      case object InvalidReceiptsType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 7
         val name: String = "InvalidReceipts"
       }
-      case object RequestFailedType extends BlacklistReasonType {
+      case object FastSyncRequestFailedType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 8
-        val name: String = "RequestFailed"
+        val name: String = "FastSyncRequestFailed"
       }
-      case object PeerActorTerminatedType extends BlacklistReasonType {
+      case object PeerActorTerminatedType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 9
         val name: String = "PeerActorTerminated"
       }
-      case object InvalidStateResponseType extends BlacklistReasonType {
+      case object InvalidStateResponseType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 10
         val name: String = "InvalidStateResponse"
       }
-      case object InvalidPivotBlockElectionResponseType extends BlacklistReasonType {
+      case object InvalidPivotBlockElectionResponseType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 11
         val name: String = "InvalidPivotElectionResponse"
       }
-      case object PivotBlockElectionTimeoutType extends BlacklistReasonType {
+      case object PivotBlockElectionTimeoutType extends BlacklistReasonType with FastSyncBlacklistGroup {
         val code: Int = 12
         val name: String = "PivotBlockElectionTimeout"
       }
+      case object DisconnectRequestedType extends BlacklistReasonType with P2PBlacklistGroup {
+        val code: Int = 13
```

Comment:
Side note : it seems that these codes would be tedious to maintain if we want to modifiy this enumeration and are not really used. We could probably remove them.

Reply:
that wasn't my idea, but it's gonna be rethought in the following ticket